### PR TITLE
PWN-8460 - Striga sms input logic

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8460

## Description of Work

- sms input logic

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt [hunk 1]
```diff
@@ -19,6 +19,7 @@ interface SmsInputContract {
         fun renderButtonLoading(isLoading: Boolean)
 
         fun navigateNext()
+        // not needed in other fragments, need to remove and create good reusable abstraction
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt:22]: I'd like you to create a separate fragment and use only .xml.

File: app/src/main/java/org/p2p/wallet/smsinput/striga/StrigaSmsInputPresenter.kt [hunk 2]
```diff
@@ -1,21 +1,101 @@
...
+) : BasePresenter<SmsInputContract.View>(), SmsInputContract.Presenter {
 
     override fun attach(view: SmsInputContract.View) {
         super.attach(view)
-        view.renderSmsTimerState(SmsInputContract.Presenter.SmsInputTimerState.ResendSmsReady)
+        view.renderSmsTimerState(SmsInputTimerState.ResendSmsReady)
     }
 
     override fun onSmsInputChanged(smsCode: String) {
+        // same logic in onboarding, need to reuse
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/smsinput/striga/StrigaSmsInputPresenter.kt:25]: no, that's good

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt (hunk 1 above)
Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt:22]: we have new design for a separate screen when the input is blocked, it appeared after i've made a task related to sms verification refactoring
image

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt:22]: I suggest redoing sms verification screen again, i didn't find it reusable, a lot of code that is inherited from BaseSmsInputFragment that is not needed.
I would rather exract all UI logic to a widget and use it on fragments needed

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt:22]: im not doing this in this task ofc

Comment (eduardmaximovich) [reply app/src/main/java/org/p2p/wallet/smsinput/SmsInputContract.kt:22]: This screen is ready #1798

File: app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt [hunk 3]
```diff
@@ -20,7 +20,7 @@ class TopUpWalletPresenter(
     override fun attach(view: TopUpWalletContract.View) {
         super.attach(view)
 
-        if (strigaSignupFeatureToggle.isFeatureEnabled && isUserAuthByWeb3) {
+        if (true) {
```
Comment (eduardmaximovich) [inline app/src/main/java/org/p2p/wallet/striga/ui/TopUpWalletPresenter.kt:23]: Don't forget ⬆️

**Pull request merged**